File: backend/app/schemas/alert.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class AlertResponse(BaseModel):
    id: str
    bid_id: str
    tender_id: Optional[str]
    tender_title: str
    deadline: Optional[datetime]
    days_left: Optional[int]
    threshold: int
    message: str
    is_read: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/alert.ts
export interface Alert {
  id: string;
  bid_id: string;
  tender_id: string | null;
  tender_title: string;
  deadline: string | null;
  days_left: number | null;
  threshold: number;
  message: string;
  is_read: boolean;
  created_at: string;
  kind: string;
  score: number | null;
}